rero_mef/api_mef.py

```diff
@@ -106,116 +94,82 @@ def get_all_pids_without_viaf(cls):
-                        .scan(),
-                        length=query.count(),
-                        verbose=verbose
-                    )
-                    for hit in progress:
-                        data = hit.to_dict()
-                        mef_pid = data['pid']
-                        agent_pid = data[agent_name]['pid']
-                        pids[record_type].setdefault(agent_pid, [])
-                        pids[record_type][agent_pid].append(mef_pid)
-                        if len(pids[record_type][agent_pid]) > 1:
-                            multiple_pids[record_type][agent_pid] = \
-                                pids[record_type][agent_pid]
-                    if len(pids[record_type]) < agent_class.count():
-                        progress = progressbar(
-                            items=agent_class.get_all_pids(),
-                            length=agent_class.count(),
-                            verbose=verbose
-                        )
-                        for pid in progress:
-                            if not pids[record_type].pop(pid, None):
-                                missing_pids[record_type].append(pid)
-                    else:
-                        pids[record_type] = {}
-                else:
-                    current_app.logger.error(
-                        f'Record type not found: {record_type}')
-        return pids, multiple_pids, missing_pids
-
-    @classmethod
-    def get_all_missing_pids(cls, record_types=None, verbose=False):
-        """Get all missing agents.
-
-        :params record_types: Record types (pid_type).
-        :param verbose: Verbose.
-        :returns: missing pids, to much pids.
-        """
-        if record_types is None:
-            record_types = []
-        missing_pids = {}
-        to_much_pids = {}
-        entity_classes = get_entity_classes()
-        used_classes = {entity_class: entity_classes[entity_class]
-                        for entity_class in entity_classes
-                        if entity_class in record_types}
+                none_pids[record_type] = []
+                sources.append(f'{agent_class.name}.pid')
+            except Exception:
+                current_app.logger.error(
+                    f'Record type not found: {record_type}')
 
-        for entity, entity_class in used_classes.items():
-            if verbose:
-                click.echo(f'Get pids from {entity} ...')
-            missing_pids[entity] = {}
-            progress = progressbar(
-                items=entity_class.get_all_pids(),
-                length=entity_class.count(),
-                verbose=verbose
-            )
-            for pid in progress:
-                missing_pids[entity][pid] = 1
-        if verbose:
-            click.echo('Get pids from MEF and calculate missing ...')
+        # Get all pids from MEF
+        date = datetime.utcnow()
+        click.echo('Get mef')
         progress = progressbar(
```

Comment:
This kind of functions should not be in the `api.py` file but in a `cli.py` file.

Reply:
I need the output sometimes in the console.